Provider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version11/GdsStatement.cs

```diff
@@ -210,11 +194,8 @@ protected override void Free(int option)
 			if (FreeNotNeeded(option))
 				return;
 
-			lock (_database.SyncObject)
-			{
-				DoFreePacket(option);
-				(Database as GdsDatabase).DeferredPackets.Enqueue(ProcessFreeResponse);
-			}
+			DoFreePacket(option);
+			(Database as GdsDatabase).DeferredPackets.Enqueue(ProcessFreeResponse);
```

Comment:
The `GdsDatabase` can be from `Version10` to `Version13` (currently), so it would have to be in every derived `GdsStatement`. I'd like to use generics somehow here. Not there yet though.